jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -499,6 +508,9 @@ public Builder withZipkinSharedRpcSpan() {
 
     public Builder withMetrics(Metrics metrics) {
       this.metrics = metrics;
+      if (this.baggageRestrictionManager instanceof DefaultBaggageRestrictionManager) {
+        this.baggageRestrictionManager = new DefaultBaggageRestrictionManager(this.metrics);
+      }
```

Comment:
This is bad because `withMetrics` now has multiple responsibilities. 
A better way to handle this is to leave `baggageRestrictionManager` uninitialized, and initialize it in `build` with the right metrics instance.